### PR TITLE
Support for pagination abstraction instead of request parameters

### PR DESCRIPTION
**Pull request by pnowy**
The custom pagable parameters (limit, offset) are replaced to the Spring custom argument resolver PageableHandlerMethodArgumentResolver

Feature #1733

### CODE REVIEW

**Comment by jdubois** [comment]
This is not working completely.

If you use the "pagination links", and create 60 items, you will see that you have the "3" and "4" links which disappear and reappear. On the first page you have "1 2 3", and if you click on "3" you will see "1 2 3 4".
So it looks like you have an offset problem somewhere.

**Comment by pnowy** [comment]
I will take a look at it and fix.

**Comment by pnowy** [comment]
Hello,

I've checked the reported issue and this is working as worked before (I mean the introduced Spring pure pagination did not change the behavior).

Please take a look at screenshots of the new solution (number means screenshot with specific page):
[1](https://cloud.githubusercontent.com/assets/3254609/9584991/79897884-5014-11e5-8ef2-86803372482c.PNG)
[2](https://cloud.githubusercontent.com/assets/3254609/9584993/7b4712bc-5014-11e5-9dfd-a959afeae240.PNG)
[3](https://cloud.githubusercontent.com/assets/3254609/9584995/7d05042e-5014-11e5-8132-f09503eb6961.PNG)
[4](https://cloud.githubusercontent.com/assets/3254609/9585001/7ea687bc-5014-11e5-8e4b-c73f192e743e.PNG)

And below you will find the screenshots with the origin generator (I've used the same data):
[1_orig](https://cloud.githubusercontent.com/assets/3254609/9585031/a2f93f24-5014-11e5-8e37-801f504f3b04.PNG)
[2_orig](https://cloud.githubusercontent.com/assets/3254609/9585032/a52a482e-5014-11e5-939e-75325f3fd738.PNG)
[3_orig](https://cloud.githubusercontent.com/assets/3254609/9585033/a7356fe0-5014-11e5-9796-5ca1e1b8b710.PNG)
[4_orig](https://cloud.githubusercontent.com/assets/3254609/9585036/a93e313c-5014-11e5-9250-c10c55aa9ecf.PNG)

I would say this is not a bug - this is a feature :) The pagination links we could divide into following buttons:

"<<" and ">>" - first and last link - visible always because the PaginationUtils generates that links always
"<" and ">" - next and previous link - visible when we have next or previous page, so as can you see for 1 we do not see the previous and for 4 we do not see next button
"currentPage-2", "currentPage-1", "currentPage+1", "currentPage+2" - visible when the current page has predecessors or successors (maximum 2 for each direction). That's why for the first page we have " **1** 2 3 " and for third page we have "1 2 **3** 4"

I was trying to keep the same logic as before. But unfortunately as I mentioned before Spring pagination resolver require the 0-indexing so I had to change a little pagination.html but it should work as before. I hope I dispel your doubts.

File: app/templates/src/main/java/package/config/_WebMvcConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+@Configuration
+public class WebMvcConfiguration extends WebMvcConfigurerAdapter {
+
+    @Bean
+    public PageableHandlerMethodArgumentResolver pageableHandlerMethodArgumentResolver() {
+        PageableHandlerMethodArgumentResolver pageableResolver = new PageableHandlerMethodArgumentResolver();
+        pageableResolver.setPageParameterName("page");
+        pageableResolver.setSizeParameterName("per_page");
+        pageableResolver.setMaxPageSize(100);
```
Comment (jdubois) [inline app/templates/src/main/java/package/config/_WebMvcConfiguration.java]: I'm going to remove this to follow our Policy 1 at http://jhipster.github.io/policies.html

**Comment by jdubois** [comment]
OK I can't reproduce the issue in fact, no worries.

I'm going to remove the WebMvcConfiguration and migrate to use the Spring parameters, so we can follow our [policy 1](http://jhipster.github.io/policies.html

**Comment by pnowy** [comment]
I was thinking about rename **per_page** argument  in order to use default Spring values but I wasn't sure whether it is used on another place or something like that. But yes - if we don't need that specific name it's a very good idea to use Spring defaults :+1:

**Pull request merged**